File: fuzz_targets/fuzz_repeat_fasta.c
// Fuzz target for repeat_fasta in benchmarks/c/fasta.c:75
// Risk Level: HIGH
// Input Types: parsing

#include <stdint.h>
#include <stddef.h>
#include <string.h>
#include <stdlib.h>

// Include the header containing repeat_fasta
// #include "benchmarks/c/fasta.h"

int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
    if (size == 0) return 0;
    
    // Null-terminate data for string functions
    char *input = malloc(size + 1);
    if (!input) return 0;
    memcpy(input, data, size);
    input[size] = '\0';
    
    // TODO: Call repeat_fasta with fuzzed input
    // Example: repeat_fasta(input);
    
    free(input);
    return 0;
}

File: fuzz_targets/cargo_fuzz/src/fuzz_repeat_fasta.rs
// Fuzz target for repeat_fasta in benchmarks/rust/fasta.rs:66
// Risk Level: CRITICAL
// Input Types: parsing, memory_buffer

#![no_main]
use libfuzzer_sys::fuzz_target;

fuzz_target!(|data: &[u8]| {
    if data.is_empty() {
        return;
    }
    
    // Convert bytes to string for parsing functions
    if let Ok(input_str) = std::str::from_utf8(data) {
        // TODO: Call repeat_fasta with fuzzed input
        // Example: repeat_fasta(input_str);
    }
});
